test/parallel/test-net-write-cb-on-destroy.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const net = require('net');
+
+const server = net.createServer();
+server.listen(0, common.mustCall(() => {
+  const socket = new net.Socket();
+
+  socket.connect({
+    port: server.address().port,
+  });
+
+  assert(socket.connecting);
+
+  socket.write('foo', common.expectsError({
+    code: 'ERR_SOCKET_CLOSED',
```

Comment:
I think this error is a bit misleading. The socket is not closed when `socket.write()` is called. Anyway this is not a blocker.